### PR TITLE
Added oscap_content related tests.

### PR DESCRIPTION
**Pull request by kbidarkar**
a) Create oscap_content.
b) Check whether Default oscap_content exists.
c) Delete oscap_content.

### CODE REVIEW

**Comment by kbidarkar** [comment]
d) Will add the `update oscap_content` testcase soon.

File: robottelo/ui/oscapcontent.py [hunk 1]
```diff
@@ -18,13 +18,18 @@ def _search_locator(self):
         """Specify locator for OpenScap content entity search procedure"""
         return locators['oscap.content_select']
 
-    def create(self, name, oscap_content_path=None):
+    def create(self, name, content_path=None):
         """Creates new oscap Content from UI"""
-        self.click(locators['oscap.new_content'])
+        self.click(locators['oscap.upload_content'])
```
Comment (kbidarkar) [inline robottelo/ui/oscapcontent.py]: Minor modifications to make the code a bit more intuitive.

**Comment by kbidarkar** [comment]
```
(robottelo)[xyz@abc robottelo]$ nosetests -s -v tests.foreman.ui.test_oscapcontent
@Test: Create OpenScap content. ... 
----------------------------------------------------------------------
Ran 3 tests in 758.658s

OK
```

**Comment by elyezer** [comment]
I don't see the file `ssg-rhel6-ds.xml` being added. Is it missing?

File: tests/foreman/ui/test_oscapcontent.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        @Feature: OpenScap - Positive Create.
+
+        @Steps:
+
+        1. Create an openscap content.
+        2. Provide all the appropriate parameters.
+
+        @Assert: Whether creating  content for OpenScap is successful.
+
```
Comment (elyezer) [inline tests/foreman/ui/test_oscapcontent.py]: This blank line is not needed. Please drop it.

File: tests/foreman/ui/test_oscapcontent.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        @Feature: OpenScap - Test Default Content.
+
+        @Steps:
+
+        1. Navigate to oscap Content page.
+        2. Set Org as Any Org.
+
+        @Assert: Whether oscap content exists by default.
+
```
Comment (elyezer) [inline tests/foreman/ui/test_oscapcontent.py]: This blank line is not needed. Please drop it.

File: tests/foreman/ui/test_oscapcontent.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        @Feature: OpenScap - Delete.
+
+        @Steps:
+
+        1. Create an openscap content.
+        2. Provide all the appropriate parameters.
+        3. Delete the openscap content.
+
+        @Assert: Deleting content for OpenScap is successful.
+
```
Comment (elyezer) [inline tests/foreman/ui/test_oscapcontent.py]: This blank line is not needed. Please drop it.

File: tests/foreman/ui/test_oscapcontent.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                        self.oscapcontent.search(content_name))
+
+    def test_positive_default_oscapcontent(self):
+        """@Test: Check whether OpenScap content exists by default.
+
+        @Feature: OpenScap - Test Default Content.
+
+        @Steps:
+
+        1. Navigate to oscap Content page.
```
Comment (elyezer) [inline tests/foreman/ui/test_oscapcontent.py]: Can you add this new module to documentation and run `make docs-clean docs` to check if it will raise any warnings related to this module?

**Comment by elyezer** [comment]
ACK, pending comments.

**Comment by sthirugn** [comment]
ACK pending @elyezer's comments

File: tests/foreman/ui/test_oscapcontent.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+class OpenScapContent(UITestCase):
+    """Implements Oscap Content tests in UI."""
+
+    @classmethod
+    def setUpClass(cls):
+        super(OpenScapContent, cls).setUpClass()
+        cls.content_path = get_data_file(OSCAP_CONTENT_FILE)
+
+    def test_positive_create_oscapcontent(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: `test_positive_create` is good enough

File: tests/foreman/ui/test_oscapcontent.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                    session.nav.go_to_oscap_content()
+                    make_oscapcontent(
+                        session,
+                        name=content_name,
+                        content_path=self.content_path,
+                    )
+                    self.assertIsNotNone(
+                        self.oscapcontent.search(content_name))
+
+    def test_positive_default_oscapcontent(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: `test_positive_check_default` sounds better

File: tests/foreman/ui/test_oscapcontent.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        """
+        with Session(self.browser) as session:
+            set_context(session, org=ANY_CONTEXT['org'])
+            session.nav.go_to_oscap_content()
+            self.assertIsNotNone(self.oscapcontent.search(
+                OSCAP_DEFAULT_CONTENT['rhel7_content']))
+            self.assertIsNotNone(self.oscapcontent.search(
+                OSCAP_DEFAULT_CONTENT['rhel6_content']))
+
+    def test_positive_delete_oscapcontent(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: `test_positive_delete` is good enough

File: tests/foreman/ui/test_oscapcontent.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+                    session.nav.go_to_oscap_content()
+                    make_oscapcontent(
+                        session,
+                        name=content_name,
+                        content_path=self.content_path,
+                    )
+                    self.assertIsNotNone(
+                        self.oscapcontent.search(content_name))
+                    self.oscapcontent.delete(content_name)
+                    self.assertIsNone(self.oscapcontent.search(content_name))
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: Delete operation do that verification automatically and do not waste 15 seconds for nothing. Please, remove that assert

File: tests/foreman/ui/test_oscapcontent.py (hunk 2 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: I have no issues dropping it, but currently this is how all the test modules are written or is it that we decided to change this from now on?

File: tests/foreman/ui/test_oscapcontent.py (hunk 6 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: ok will update, I was thinking about the existing screenshot functionality. But I think we can work on that to update it and make use of the Class Name for it to categorize things when all the testcases from different module will have the same name like `test_positive_create`.

File: tests/foreman/ui/test_oscapcontent.py (hunk 9 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: oh! yeah nice that this is implicit now :+1: , missed to remove it, this code was lying since a long time , while I had been working on other stuff.

File: tests/foreman/ui/test_oscapcontent.py (hunk 2 above)
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: We started to use new approach, probably week or two ago

File: tests/foreman/ui/test_oscapcontent.py (hunk 9 above)
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: Thanks

File: tests/foreman/ui/test_oscapcontent.py (hunk 6 above)
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: Huh, it can be a very important issue. Let's me bring this up in our general discussion. Thanks

File: tests/foreman/ui/test_oscapcontent.py (hunk 2 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: ok, then updating it.

File: tests/foreman/ui/test_oscapcontent.py (hunk 5 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: I have run `make docs-clean docs` and checked for oscapcontent related WARNINGS. I fixed whichever I could, but there are still 4 more warnings.

@anyone: Please feel free to fix it, in another PR if you can quickly. :)


```
(robottelo)[xyz@abc robottelo]$ make docs-clean docs
make[1]: Entering directory '~/git_world/robottelo/docs'
rm -rf _build/*
make[1]: Leaving directory '~/git_world/robottelo/docs'
make[1]: Entering directory '~/git_world/robottelo/docs'
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.3.1
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 19 source files that are out of date
updating environment: 19 added, 0 changed, 0 removed
reading sources... [100%] reviewing_PRs                                                                                                                                                                            
~/git_world/robottelo/robottelo/helpers.py:docstring of robottelo.helpers.install_katello_ca:4: WARNING: Field list ends without a blank line; unexpected unindent.
~/git_world/robottelo/robottelo/helpers.py:docstring of robottelo.helpers.remove_katello_ca:4: WARNING: Field list ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] reviewing_PRs                                                                                                                                                                             
~/git_world/robottelo/robottelo/datafactory.py:docstring of robottelo.datafactory.invalid_values_list:None: WARNING: more than one target found for cross-reference u'list': robottelo.cli.docker.DockerRegistry.list, robottelo.cli.docker.DockerImage.list, robottelo.cli.docker.DockerTag.list, robottelo.cli.lifecycleenvironment.LifecycleEnvironment.list, robottelo.cli.docker.DockerContainer.list, robottelo.cli.base.Base.list
~/git_world/robottelo/docs/api/tests.foreman.cli.rst:37: WARNING: py:class reference target not found: Domain
generating indices... genindex py-modindex
highlighting module code... [100%] robottelo.datafactory                                                                                                                                                           
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 4 warnings.

Build finished. The HTML pages are in _build/html.
make[1]: Leaving directory '~/git_world/robottelo/docs'
```

Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: @abalakh can you help with `katello...` part?

File: robottelo/cli/factory.py [hunk 10]
```diff
@@ -1171,14 +1171,17 @@ def make_usergroup(options=None):
...
 
-    Options:
+    Options::
+
         --auth-source-id AUTH_SOURCE_ID           ID of linked auth source
         --name NAME                               External user group name
         --user-group, --usergroup USER_GROUP_NAME Name to search by
         --user-group-id, --usergroup-id USER_GROUP_ID
+
     """
```
Comment (kbidarkar) [inline robottelo/cli/factory.py]: @sghai , these changes are to reduce the `warnings` from `make docs-clean docs`.

I know this should be in a different PR, but I could fix it so it's here.

File: robottelo/ui/container.py [hunk 11]
```diff
@@ -144,7 +144,7 @@ def delete(self, resource_name, container_name, really=True):
     def set_power_status(self, resource_name, cont_name, power_on):
         """Perform power on or power off for container
 
-        :param boolean power_on: True - for On, False - for Off
+        :param bool power_on: True - for On, False - for Off
```
Comment (kbidarkar) [inline robottelo/ui/container.py]: @sghai, Fixed this to reduce the warnings related to the `make docs-clean docs`.

**Comment by kbidarkar** [comment]
```
(robottelo)[xyz@abc robottelo]$ nosetests -s -v tests.foreman.ui.test_oscapcontent
@Test: Create OpenScap content with negative values ...
======================================================================
.
.
. (truncated) as it's intense  logging.
ERROR: @Test: Create OpenScap content.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kbidarka/git_world/robottelo/tests/foreman/ui/test_oscapcontent.py", line 49, in test_positive_create
    self.oscapcontent.search(content_name))
  File "/home/kbidarka/git_world/robottelo/robottelo/ui/base.py", line 88, in search
    self.navigate_to_entity()
  File "/home/kbidarka/git_world/robottelo/robottelo/ui/oscapcontent.py", line 15, in navigate_to_entity
    Navigator(self.browser).go_to_oscap_content()
  File "/home/kbidarka/git_world/robottelo/robottelo/ui/navigator.py", line 375, in go_to_oscap_content
    menu_locators['menu.oscap_content'],
  File "/home/kbidarka/git_world/robottelo/robottelo/ui/navigator.py", line 14, in menu_click
    self.click(sub_menu_locator)
  File "/home/kbidarka/git_world/robottelo/robottelo/ui/base.py", line 549, in click
    .format(type(self).__name__, locator)
UINoSuchElementError: Navigator: element with locator ('xpath', "//div[contains(@style,'static') or contains(@style, 'fixed')]//a[@id='menu_item_compliance_contents']") not found while trying to click

----------------------------------------------------------------------
Ran 5 tests in 685.504s

FAILED (SKIP=1, errors=1)
```
Running the failed test again.

```
(robottelo)[xyz@abc robottelo]$ nosetests -s -v -m test_positive_create tests.foreman.ui.test_oscapcontent
@Test: Create OpenScap content. ... 
----------------------------------------------------------------------
Ran 1 test in 235.238s

OK
```

**Comment by kbidarkar** [comment]
Looks like I am on slow network and accessing the machines over long distances for the above errors. But it runs fine when re-running it.

**Comment by kbidarkar** [comment]
Needs few more changes, updating.

File: robottelo.properties.sample [hunk 12]
```diff
@@ -132,6 +132,9 @@ ssh_key=
 # bridge=BRIDGE_VALUE
 # gateway=GATEWAY_VALUE
 
+# For OSCAP Testing
+# [oscap]
+# content_path=~/ssg-rhel6-ds.xml
```
Comment (kbidarkar) [inline robottelo.properties.sample]: I believe once I start to work on policies, we can add more stuff under [oscap] section.

File: robottelo/cli/factory.py [hunk 13]
```diff
@@ -1171,14 +1171,17 @@ def make_usergroup(options=None):
...
         hammer user-group external create [OPTIONS]
 
-    Options:
+    Options::
+
         --auth-source-id AUTH_SOURCE_ID           ID of linked auth source
         --name NAME                               External user group name
         --user-group, --usergroup USER_GROUP_NAME Name to search by
         --user-group-id, --usergroup-id USER_GROUP_ID
+
```
Comment (elyezer) [inline robottelo/cli/factory.py]: This blank line should not be necessary though.

Comment (kbidarkar) [inline robottelo/cli/factory.py]: I totally agree, Let me try removing it, had to add it for `make docs-clean docs` to PASS, if not I really don't know how do I fix this. Let me re-run it with removed and the exact error it throws for this.

Comment (kbidarkar) [inline robottelo/cli/factory.py]: This error is gone for good, had to fight with it to get rid of it earlier.

**Comment by kbidarkar** [comment]
This PR is ready for review, please review again as many new test-cases have been added to it, after the initial ACK.

Also the new entry for [oscap] for robottelo-config is added to jenkins. please check for the same to review.

File: robottelo.properties.sample (hunk 12 above)
Comment (kbidarkar) [inline robottelo.properties.sample]: Why is this part of intrastructure?
Initially added the .xml file to robottelo code base , but that would bring in 44000+ lines of code to the robottelo code base. So decided to make this part of Infrastructure.

**Comment by elyezer** [comment]
The decrease of coverage is not significant right now.

**Comment by elyezer** [comment]
ACK

@oshtaier @abalakh can you please merge this if you think it is ok?

File: tests/foreman/ui/test_oscapcontent.py [hunk 14]
```diff
@@ -0,0 +1,161 @@
+from fauxfactory import gen_string
+from robottelo.config import settings
+from robottelo.constants import (
+    ANY_CONTEXT,
+    OSCAP_DEFAULT_CONTENT,
+)
+from robottelo.datafactory import valid_data_list, invalid_values_list
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: `from robottelo.datafactory import invalid_values_list, valid_data_list`

File: tests/foreman/ui/test_oscapcontent.py [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+from robottelo.decorators import tier1, tier2, skip_if_bug_open
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: `from robottelo.decorators import skip_if_bug_open, tier1, tier2`

File: tests/foreman/ui/test_oscapcontent.py [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+from robottelo.helpers import get_data_file
+from robottelo.test import UITestCase
+from robottelo.ui.factory import set_context, make_oscapcontent, make_org
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: `make_org, make_oscapcontent, set_context`

File: tests/foreman/ui/test_oscapcontent.py [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+    @tier1
+    def test_positive_default(self):
+        """@Test: Check whether OpenScap content exists by default.
+
+        @Feature: OpenScap - Test Default Content.
+
+        @Steps:
+
+        1. Navigate to oscap Content page.
+        2. Set Org as Any Org.
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: It seems that we are doing `Set Org as Any Org` part first, so it is probably:
```
1) Set Org as Any Org
2) Navigate to oscap Content page.
```

File: tests/foreman/ui/test_oscapcontent.py [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
+        2. Provide all the appropriate parameters.
+
+        @Assert: Whether creating  content for OpenScap is successful.
+        """
+        with Session(self.browser) as session:
+            for content_name in valid_data_list():
+                with self.subTest(content_name):
+                    set_context(session, org=ANY_CONTEXT['org'])
+                    session.nav.go_to_oscap_content()
+                    make_oscapcontent(
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: please, use `self.navigate_to_entity()` method instead

File: tests/foreman/ui/test_oscapcontent.py [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+        1. Create an openscap content.
+        2. Provide all the appropriate parameters.
+
+        @Assert: Whether creating  content for OpenScap is not successful.
+        """
+        with Session(self.browser) as session:
+            for content_name in invalid_values_list(interface='ui'):
+                with self.subTest(content_name):
+                    set_context(session, org=ANY_CONTEXT['org'])
+                    session.nav.go_to_oscap_content()
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: please, use `self.navigate_to_entity()` method instead

File: tests/foreman/ui/test_oscapcontent.py [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+        @Steps:
+
+        1. Navigate to oscap Content page.
+        2. Set Org as Any Org.
+
+        @Assert: Whether oscap content exists by default.
+        """
+        with Session(self.browser) as session:
+            set_context(session, org=ANY_CONTEXT['org'])
+            session.nav.go_to_oscap_content()
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: Search functionality usually does navigation, so that line seems as redundant one

File: tests/foreman/ui/test_oscapcontent.py [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+        2. Provide all the appropriate parameters.
+        3. Update the openscap content, here the Org.
+
+        @Assert: Whether creating  content for OpenScap is successful.
+        """
+        org = gen_string('alpha')
+        content_name = gen_string('alpha')
+        with Session(self.browser) as session:
+            set_context(session, org=ANY_CONTEXT['org'])
+            session.nav.go_to_oscap_content()
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: As far as I see make_oscapcontent() does navigation by itself, so please remove that line

File: tests/foreman/ui/test_oscapcontent.py [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+            )
+            make_org(
+                session,
+                org_name=org,
+            )
+            self.assertIsNotNone(
+                self.org.search(org))
+            self.oscapcontent.update(content_name, content_org=org)
+            session.nav.go_to_select_org(org)
+            session.nav.go_to_oscap_content()
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: Search functionality usually does navigation, so that line seems as redundant one

File: tests/foreman/ui/test_oscapcontent.py [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+        2. Provide all the appropriate parameters.
+        3. Delete the openscap content.
+
+        @Assert: Deleting content for OpenScap is successful.
+        """
+        with Session(self.browser) as session:
+            for content_name in valid_data_list():
+                with self.subTest(content_name):
+                    set_context(session, org=ANY_CONTEXT['org'])
+                    session.nav.go_to_oscap_content()
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: As far as I see make_oscapcontent() does navigation by itself, so please remove that line

File: tests/foreman/ui/test_oscapcontent.py [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+        content_name = gen_string('alpha')
+        with Session(self.browser) as session:
+            set_context(session, org=ANY_CONTEXT['org'])
+            session.nav.go_to_oscap_content()
+            make_oscapcontent(
+                session,
+                name=content_name,
+                content_path=self.content_path,
+            )
+            make_org(
```
Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: Maybe, we can use nailgun for that purpose as we usually do in other UI code:
```
org = entities.Organization(name=org_name).create()
```
so we can remove:
`org = gen_string('alpha')`
and use:
```
self.oscapcontent.update(content_name, content_org=org.name)
session.nav.go_to_select_org(org.name)
```

**Comment by oshtaier** [comment]
pending comments

File: tests/foreman/ui/test_oscapcontent.py (hunk 21 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: Not sure why I was using this all the while.

**Comment by oshtaier** [comment]
Maybe re-run scripts once more after all comments addressed. Please:)

File: tests/foreman/ui/test_oscapcontent.py (hunk 24 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscapcontent.py]: I do want to use "alpha" string only, unless it's a very basic test.

Comment (oshtaier) [inline tests/foreman/ui/test_oscapcontent.py]: so `org = entities.Organization(name=gen_string('alpha')).create()` should suffice your request

File: tests/foreman/ui/test_oscapcontent.py (hunk 5 above)
Comment (abalakh) [inline tests/foreman/ui/test_oscapcontent.py]: yep, will fix those

**Comment by kbidarkar** [comment]
```
(robottelo)[xyz@abc robottelo]$ nosetests -s -v tests.foreman.ui.test_oscapcontent
@Test: Create OpenScap content with negative values ...

----------------------------------------------------------------------
Ran 5 tests in 743.666s

OK (SKIP=1)
```
Ran the above tests after fixing all for comments.

**Comment by oshtaier** [comment]
ACK

**Comment by oshtaier** [comment]
please rebase
